### PR TITLE
refactor(api): decompose model service and reinitialize routes

### PR DESCRIPTION
**Pull request by 1larity**
## Summary
This PR decomposes model-service and reinitialize API routes out of `acestep/api_server.py` into focused modules while preserving legacy client behavior.

The branch now includes:
- route registration + inventory logic in `acestep/api/http/model_service_routes.py`
- blocking model initialization logic in `acestep/api/http/model_init_service.py`
- reinitialize endpoint wiring in `acestep/api/http/reinitialize_route.py`
- targeted unit + HTTP tests for each extracted module

### In scope
- decomposition of `/health`, `/v1/stats`, `/v1/models`, `/v1/model_inventory`, `/v1/init`, and `/v1/reinitialize`
- preserving existing response/status/error contracts for API clients
- targeted regression tests for contract parity

### Out of scope
- endpoint contract redesign/cleanup
- changing legacy wrapper/status semantics
- unrelated API/training/generation behavior

## What Changed
- `acestep/api/http/model_service_routes.py` (new)
  - extracted route registration for health/stats/model inventory/init
  - extracted `_collect_model_inventory` helper used by model endpoints
- `acestep/api/http/model_init_service.py` (new)
  - extracted blocking DiT/LM init path used by `/v1/init`
  - kept legacy state mutation/error semantics
- `acestep/api/http/reinitialize_route.py` (new)
  - extracted `/v1/reinitialize` endpoint logic
  - preserved legacy raw 500 vs wrapped-response behavior edge cases
- `acestep/api_server.py`
  - removed inline implementations and replaced with registration calls

## Review Concerns and Resolutions
1. **Concern: preserve client contracts even if legacy behavior is inconsistent**
   - **Resolved:** contracts intentionally preserved. No endpoint semantic cleanup in this decomposition.
2. **Concern: `/v1/init` error contract drift risk**
   - **Resolved:** preserved wrapped error envelope behavior (`HTTP 200` + `code=500` on init failures).
3. **Concern: `/v1/reinitialize` legacy edge-case behavior drift risk**
   - **Resolved:** preserved raw `HTTP 500` when handler missing; preserved wrapped success path when `llm_handler` missing.
4. **Concern: permission-sensitive test filesystem access**
   - **Resolved:** tests use non-writing paths and explicit `os.makedirs` mocking where needed.

### Automated (all passed)
- `.\.venv\Scripts\python.exe -m unittest -v acestep.api.http.model_service_routes_http_test acestep.api.http.model_init_service_test acestep.api.http.model_service_routes_test acestep.api.http.reinitialize_route_test acestep.api.http.reinitialize_route_http_test`
  - **Ran 21 tests: OK**

## Risk / Compatibility
- This PR is decomposition-only for the targeted routes.
- Existing client contracts are intentionally preserved, including known legacy inconsistencies.
- If legacy behavior is poor quality, it remains unchanged in this slice by design.

## @coderabbit review focus
Please prioritize **behavioral parity and client contract preservation** over stylistic modernization in this PR.
- Verify response shapes, status-code semantics, and error wrapping match prior behavior.
- Flag only true behavioral drift; avoid recommending contract changes in this decomposition slice unless they are regressions.

### CODE REVIEW

File: acestep/api/jobs/store.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        """Mark a job as failed and persist an error message."""
+        with self._lock:
+            record = self._jobs[job_id]
+            record.status = "failed"
+            record.finished_at = time.time()
+            record.result = None
+            record.error = error
+            record.progress = record.progress if record.progress > 0 else 0.0
+            record.stage = "failed"
+            record.updated_at = time.time()
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/api/jobs/store.py:146]: Is this a pre existing behaviour from the decomposition target?

File: acestep/ui/gradio/events/results/batch_management_background.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        raw_codes_list = final_result[47] if len(final_result) > 47 else [""] * 8
+        generated_codes_batch = raw_codes_list if isinstance(raw_codes_list, list) else [""] * 8
+        generated_codes_single = generated_codes_batch[0] if generated_codes_batch else ""
+        extra_outputs_from_bg = final_result[46] if len(final_result) > 46 and final_result[46] is not None else {}
+        scores_from_bg = _extract_scores(final_result)
+
+        batch_size = params.get("batch_size_input", 2)
+        allow_lm_batch_val = params.get("allow_lm_batch", False)
+        if allow_lm_batch_val and batch_size >= 2:
+            codes_to_store = generated_codes_batch[:int(batch_size)]
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/ui/gradio/events/results/batch_management_background.py:126]: Is this a pre existing behaviour from the decomposition target?

**Comment by 1larity** [comment]
Several @coderabbit comments cover pre-existing bugs/poor design, I have honoured existing API behaviour so as to not break existing API clients.

**Comment by 1larity** [comment]
Addressed the substantive open review concerns in this PR update:)
 /v1/audio hardening
   - switched to Path.resolve + relative_to boundary checks
   - added explicit file-only guard (directory targets now return 404 before FileResponse)
   -  Cross-platform deterministic audio tests
   - removed Windows-specific path assumption in unit test
   - added directory-target regression test
   -  release_task request builder override collision
   - changed builder to construct payload dict then apply overrides before model construction\
   -  prevents duplicate-kwarg TypeError when overrides include canonical fields
   - added regression test for generic override collision/precedence
 
Validation run:
   - .venv\\Scripts\\python.exe -m unittest -v acestep.api.http.audio_route_test
   -  .venv\\Scripts\\python.exe -m unittest -vacestep.api.http.audio_route_http_test acestep.api.http.release_task_request_builder_test
   - .venv\\Scripts\\python.exe -m unittest -v acestep.api.http.sample_format_routes_test
   -  .venv\\Scripts\\python.exe -m unittest -v acestep.api.http.sample_format_routes_http_test
   -  .venv\\Scripts\\python.exe -m unittest -v acestep.api.http.audio_route_test
   -  .venv\\Scripts\\python.exe -m unittest -v acestep.api.http.audio_route_http_test
   -  .venv\\Scripts\\python.exe -m unittest -v acestep.api.http.query_result_route_test
   -  .venv\\Scripts\\python.exe -m unittest -v acestep.api.http.query_result_route_http_test
   -  .venv\\Scripts\\python.exe -m unittest -v acestep.api.http.release_task_request_builder_test
   -  .venv\\Scripts\\python.exe -m unittest -v acestep.api.http.release_task_request_parser_test
   -  .venv\\Scripts\\python.exe -m unittest -v acestep.api.http.release_task_route_http_test\n\nAll tests passed.

File: acestep/api/http/audio_route.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            raise HTTPException(status_code=404, detail="Audio file not found")
+
+        ext = resolved_path.suffix.lower()
+        media_types = {
+            ".mp3": "audio/mpeg",
+            ".wav": "audio/wav",
+            ".flac": "audio/flac",
+            ".ogg": "audio/ogg",
+        }
+        return FileResponse(str(resolved_path), media_type=media_types.get(ext, "audio/mpeg"))
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/api/http/audio_route.py:39]: Pre-existing possibly deliberate. The GHAS taint warnings are not because decomposition invented new behaviour; decomposition mainly moved that logic into [audio_route.py] .

File: acestep/api/http/audio_route.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+        from fastapi.responses import FileResponse
+
+        resolved_path = Path(path).resolve(strict=False)
+        allowed_dir = Path(request.app.state.temp_audio_dir).resolve(strict=False)
+        try:
+            resolved_path.relative_to(allowed_dir)
+        except ValueError:
+            raise HTTPException(status_code=403, detail="Access denied: path outside allowed directory")
+        if not resolved_path.is_file():
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/api/http/audio_route.py:29]: The GHAS taint warnings are not because decomposition invented new behaviour; decomposition mainly moved that logic into [audio_route.py] (and then I hardened it).

File: acestep/api/http/audio_route.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+) -> None:
+    """Register the ``GET /v1/audio`` route."""
+
+    @app.get("/v1/audio")
+    async def get_audio(path: str, request: Request, _: None = Depends(verify_api_key)):
+        """Serve a generated audio file when path is within the allowed directory."""
+
+        from fastapi.responses import FileResponse
+
+        resolved_path = Path(path).resolve(strict=False)
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/api/http/audio_route.py:23]: @ChuxiJ  I have marked as resolved, this was pre-existing behaviour, I can see that this may be desirable in some circumstances, though maybe it could be handled better. I did not want to break, if this is the intended behaviour.

**Comment by 1larity** [comment]
@ChuxiJ Can you instruct me how you wish to proceed with the file handling security issues in both PRs:-

A) Retain current behaviour:- Accept as is, attach clear warning to installer "Client can specify paths outside the  project space". You would need to add an exception to Github bot to ignore this issue.

B) Fix to use safe path (possible client app breaking change).

**Comment by ChuxiJ** [comment]
> @ChuxiJ Can you instruct me how you wish to proceed with the file handling security issues in both PRs:-
> 
> A) Retain current behaviour:- Accept as is, attach clear warning to installer "Client can specify paths outside the project space". You would need to add an exception to Github bot to ignore this issue.
> 
> B) Fix to use safe path (possible client app breaking change).

-> A

**Comment by 1larity** [comment]
@ChuxiJ rebased to latest.

**Review by ChuxiJ — APPROVED** [review]
